### PR TITLE
Added Panel & Panel Row components to DSM 

### PR DESCRIPTION
**Pull request by OwenCoogan**
### What does this PR do?

Related to: #

https://linear.app/upfluence/issue/ENG-1088/[oss-components]-osspanel-and-osspanelrow-components

### What are the observable changes?

Added 2 new components : OSS::Panel & OSS::Panel::Row . 

Capture d’écran 2023-09-05 à 11 18 33

Here with hover state on Panel Row

Capture d’écran 2023-09-05 à 11 18 37
<br/>

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: tests/integration/components/o-s-s/panel-test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  test('it renders', async function (assert) {
+    await render(hbs`
+      <OSS::Panel>
+        <:content>Content</:content>
+      </OSS::Panel>
+    `);
+    assert.dom('.oss-panel').exists();
+    assert.dom('.oss-panel--content').hasText('Content', 'Content should be visible');
+    assert.dom('.oss-panel--footer').doesNotExist('Footer should not be visible');
```
Comment (Miexil) [inline tests/integration/components/o-s-s/panel-test.ts:17]: What does the doesNotExist parameter do ? 🤔

File: tests/integration/components/o-s-s/panel-test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    assert.dom('.oss-panel--header').doesNotExist('Header should not be visible');
+  });
+  test('it should render content separator if header exists', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/panel-test.ts:20]: line-break between tests

Comment (Miexil) [reply tests/integration/components/o-s-s/panel-test.ts:20]: it should render *the* content separator if *the* header exists

(same below)

File: tests/integration/components/o-s-s/panel-test.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  test('it should not render footer separator if content does not exist', async function (assert) {
+    await render(hbs`
+      <OSS::Panel>
+        <:footer>Footer</:footer>
+      </OSS::Panel>
+    `);
+    assert
+      .dom('.oss-panel--footer .oss-panel--separator')
+      .doesNotExist('Footer separator should not be visible when content is not visible');
+  });
```
Comment (Miexil) [inline tests/integration/components/o-s-s/panel-test.ts:63]: Add basic tests that verify the named-blocks, eg:
- If the header named-block is defined, it is properly rendered
- If the content named-block is defined, it is properly rendered
- If the footer named-block is defined, it is properly rendered

File: app/styles/organisms/panel.less [hunk 4]
```diff
@@ -0,0 +1,60 @@
+.oss-panel {
+  border-radius: var(--radius-border-radius-md, 8px);
+  border: 1px solid var(--color-gray-200, #E5E7EB);
+  background: var(--color-default-white, #FFF);
```
Comment (JulienVannier66) [inline app/styles/organisms/panel.less:4]:
```suggestion
  border-radius: var(--radius-border-radius-md);
  border: 1px solid var(--color-border-default);
  background: var(--color-white);
```
You don't need to set default value when you are using `var`, also `--color-default-white` variable doesn't exist

File: app/styles/organisms/panel.less [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    display: inline-flex;
+    align-items: center;
+    cursor: pointer;
+    height: 36px;
+    width: 100%;
+    padding: var(--spacing-px-9) var(--spacing-px-18);
+    align-items: center;
+    gap: var(--spacing-px-9);
+    flex-shrink: 0;
+    background: var(--color-default-white);
```
Comment (JulienVannier66) [inline app/styles/organisms/panel.less:33]:
```suggestion
    background: var(--color-white);
```

File: app/styles/organisms/panel.less [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    p {
+      margin-bottom: 0;
+    }
+  }
+
+  .oss-panel--separator {
+    margin: 0;
+    border-top: 1px solid var(--color-gray-200);
```
Comment (JulienVannier66) [inline app/styles/organisms/panel.less:57]:
```suggestion
    border-top: 1px solid var(--color-border-default);
```

File: tests/integration/components/o-s-s/panel-row-test.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<OSS::Panel::Row @icon={{"fa-cog"}} @label={{'Your Label'l}} />`);
+
+    assert.dom('.oss-panel-content--row').exists();
+    assert.dom('.far').exists();
+    assert.dom('p').hasText('Your Label');
+  });
+});
```
[the remark this answers is not in the document]
Comment (JulienVannier66) [reply tests/integration/components/o-s-s/panel-row-test.ts:16]: Split your test for these asserts as Max says
```
assert.dom('.far').exists();
assert.dom('p').hasText('Your Label');
```

File: tests/integration/components/o-s-s/panel-test.ts (hunk 1 above)
Comment (JulienVannier66) [reply tests/integration/components/o-s-s/panel-test.ts:17]: Display an extra message in the test next to `OK` if I remember

File: addon/components/o-s-s/panel/row.ts
Comment (aprentout) [inline addon/components/o-s-s/panel/row.ts:1]: If one argument is required we can also add 'assertion' to check validity on runtime ([example](https://github.com/upfluence/oss-components/blob/57d66b0e241563354b0f14a1584bfa67081dca96/addon/components/o-s-s/button.ts#L96))

File: tests/integration/components/o-s-s/panel-test.ts (hunk 1 above)
Comment (OwenCoogan) [reply tests/integration/components/o-s-s/panel-test.ts:17]: It shouldn't be in this test but I wanted to check if footer didn't exist based on what named blocks were present in this test case. I test the name blocks further down.

File: addon/components/o-s-s/panel/row.ts
Comment (OwenCoogan) [reply addon/components/o-s-s/panel/row.ts:1]: @aprentout talked about it with max & for now we'll pass that argument as optional. This subject raises a potential task I suggested , we can talk about it after

File: app/styles/organisms/panel.less [hunk 8]
```diff
@@ -0,0 +1,44 @@
+.oss-panel {
+  border-radius: var(--radius-border-radius-md);
```
Comment (JulienVannier66) [inline app/styles/organisms/panel.less:2]:
```suggestion
  border-radius: var(--border-radius-md);
```

File: tests/integration/components/o-s-s/panel-row-test.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<OSS::Panel::Row @icon={{"fa-cog"}} @label={{'Your Label'l}} />`);
+    assert.dom('.oss-panel-content--row').exists('OK');
+  });
+
+  test('It renders the @ICON properly', async function (assert) {
+    await render(hbs`<OSS::Panel::Row @icon={{"fa-cog"}} @label={{'Your Label'l}} />`);
+    assert.dom('.far').exists('OK');
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/panel-row-test.ts:16]:
```suggestion
    assert.dom('.far').exists(');
```
Same here

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: tests/integration/components/o-s-s/panel-row-test.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+module('Integration | Component | o-s-s/panel-row', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<OSS::Panel::Row @icon={{"fa-cog"}} @label={{'Your Label'l}} />`);
+    assert.dom('.oss-panel-content--row').exists('OK');
+  });
+
+  test('It renders the @ICON properly', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/panel-row-test.ts:14]: nit: @icon is lowercase

File: tests/integration/components/o-s-s/panel-row-test.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    await render(hbs`<OSS::Panel::Row @icon={{"fa-cog"}} @label={{'Your Label'l}} />`);
+    assert.dom('.far').exists('OK');
+  });
+
+  test('it renders the @Label properly', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/panel-row-test.ts:19]: nit: @label is lowercase

**Review by Miexil — COMMENTED** [review]
Minor issues left 😇

**Review by Miexil — APPROVED** [review]
7y3dd0

**Pull request merged**